### PR TITLE
add verify prod and refactor terraform check/deploy

### LINKED ISSUE #54: Setup a "plan" presubmit
This needs two separate PRS.

1. We should create a NEW identity in `iac/bootstrap` that's a project viewer and can be federated by pull requests presubmit (e.g. `ref:refs/heads/main` -> `:pull_request`). -- I may need to apply this once it lands, or add folks to ACLs.

1. We should add a presubmit like mono that posts the plan to the PR body.

cc @cpanato

### PR DESCRIPTION
**Pull request by cpanato**
Fixes #54

Need https://github.com/octo-sts/app/pull/196

### CODE REVIEW

File: .github/workflows/verify-prod.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Copyright 2024 Chainguard, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+name: Verify prod Octo-sts
+
+on:
+  pull_request_target:
+    branches:
+      - 'main'
```
Comment (mattmoor) [inline .github/workflows/verify-prod.yaml:9]: I think we want to add a labeling requirement here as well, since this repo is now public.

Comment (cpanato) [reply .github/workflows/verify-prod.yaml:9]: oh, ok, so we will need to trigger when is updated as well

Comment (cpanato) [reply .github/workflows/verify-prod.yaml:9]: are we doing a similar thing in another repo? or will this be the first one?

Comment (mattmoor) [reply .github/workflows/verify-prod.yaml:9]: we should, but not yet

File: .github/workflows/.terraform.yaml [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+      - name: Terraform init
+        id: init
+        run: terraform init -reconfigure
+
+      - name: Terraform Validate
+        id: validate
+        run: terraform validate -no-color
+
+      - name: Terraform Plan
```
Comment (mattmoor) [inline .github/workflows/.terraform.yaml:103]: Do you know of anything that we can run over this to check for injection vulnerabilities?

Skimming, the only one that stood out is `inputs.working_directory`, and that's a reach, but would be good to sanity check things

**Review by mattmoor — APPROVED** [review]

**Comment by mattmoor** [comment]
I'm going to merge this, so that we can start using it (probably would have avoided mild outage yesterday).  I'd love an answer for the shell check issue.

**Review by SamyDgz — APPROVED** [review]

**Pull request merged**